binnacle/lib/ui/navigation/panel.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:sliding_up_panel/sliding_up_panel.dart';
+import 'package:sos/bloc.dart';
+
+import 'package:sos/enums.dart';
+import 'package:sos/providers/app_provider.dart';
+import 'package:sos/providers/navigation_provider.dart';
+import 'package:sos/ui/global_theme.dart';
+
+class NavigationPanel extends StatelessWidget {
+  NavigationPanel({Key key}) : super(key: key);
+
+  Bloc bloc;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = GlobalTheme();
+    bloc = Provider.of(context);
+
+    return StreamBuilder(
+        stream: bloc.navigationEventBus,
+        builder: (BuildContext context, AsyncSnapshot snapshot) {
+          if (snapshot.hasError || !snapshot.hasData || snapshot.data.eventType == null) {
```

Comment:
Should at least have some print statement. As is it's like an empty catch

Reply:
This stream gets rebuilt several times. I found it better to print in the Widget _panel code in the switch statement. Cut down a lot of the print statements and only showed the ones we might care about. Let me know if that's okay.